### PR TITLE
BUG:UPHRH 3852, 3806,3807 assesment warning message and best score issue resolved

### PR DESCRIPTION
**Pull request by sohailamjad12**
# SunbirdEd - Portal
---

##assesment warning message and best score issue resolved
---

#### Example
- [x] Applicable
- [ ] Not applicable

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts [hunk 1]
```diff
@@ -118,6 +118,7 @@ export class AssessmentPlayerComponent implements OnInit, OnDestroy, ComponentCa
   contentRatingModal = false;
   isDesktopApp= false;
   isConnected= true;
+  assessmenType:any;
```
Comment (devendra-tarento) [inline src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts]: @sohailamjad12 rename variable name to isPIAA (boolean variable) as you are saving boolean value to it not assessmentType (PIAA assessment, Self assessment etc);

File: src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts [hunk 2]
```diff
@@ -925,10 +927,10 @@ export class AssessmentPlayerComponent implements OnInit, OnDestroy, ComponentCa
   }
 
   onSelfAssessLastAttempt(event) {
-    if (_.get(event, 'data') === 'renderer:selfassess:lastattempt' || _.get(event, 'edata.isLastAttempt')) {
+    if (_.get(event, 'data') === 'renderer:selfassess:lastattempt' || _.get(event, 'edata.isLastAttempt' && !this.assessmenType)) {
```
Comment (devendra-tarento) [inline src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts]: @sohailamjad12 If first condition true then if block will execute without checking assessmentType. Second condition after or operator is not correct.

File: src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts [hunk 3]
```diff
@@ -925,10 +927,10 @@ export class AssessmentPlayerComponent implements OnInit, OnDestroy, ComponentCa
...
       this.toasterService.error(_.get(this.resourceService, 'frmelmnts.lbl.selfAssessLastAttempt'));
     }
-    if (_.get(event, 'data') === 'renderer:maxLimitExceeded' || _.get(event, 'edata.maxLimitExceeded')) {
+    if (_.get(event, 'data') === 'renderer:maxLimitExceeded' || _.get(event, 'edata.maxLimitExceeded') && !this.assessmenType) {
```
Comment (devendra-tarento) [inline src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts]: @sohailamjad12 If first condition true then if block will execute without checking assessmentType

File: src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts [hunk 4]
```diff
@@ -94,6 +94,7 @@ export class CoursePlayerComponent implements OnInit, OnDestroy {
   isConnected = false;
   dropdownContent = true;
   showForceSync = true;
+  primaryCategoryType:any;
```
Comment (devendra-tarento) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts]: @sohailamjad12   rename variable name to isPrimaryCategoryPIAA (boolean variable) as you are saving boolean value.

File: src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts [hunk 5]
```diff
@@ -343,7 +344,10 @@ export class CoursePlayerComponent implements OnInit, OnDestroy {
       .subscribe((res) => {
         const _parsedResponse = this.courseProgressService.getContentProgressState(req, res);
         this.progressToDisplay = Math.floor((_parsedResponse.completedCount / this.courseHierarchy.leafNodesCount) * 100);
-        this.contentStatus = _parsedResponse.content || [];
+        this.primaryCategoryType = (_.get(this.courseHierarchy, 'primaryCategory')) === 'PIAA Assessment';
```
Comment (devendra-tarento) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts]: @sohailamjad12 is this change to not display Best Score or assessment Progress ?

File: src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts (hunk 1 above)
Comment (sohailamjad12) [reply src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts]: resolved

File: src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts (hunk 2 above)
Comment (sohailamjad12) [reply src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts]: Resolved

File: src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts (hunk 4 above)
Comment (sohailamjad12) [reply src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts]: resolved

**Pull request merged**